nemo/collections/nlp/models/machine_translation/megatron_nmt_model.py

```diff
@@ -801,3 +977,16 @@ def on_validation_start(self) -> None:
 
     def on_test_start(self) -> None:
         self.trainer.test_loop._data_fetcher = GlobalBatchDataFetcher()
+
+    # the following properties are needed for nemo2riva tool to work
+    @property
+    def encoder(self):
+        return EncWEmb(self.enc_dec_model.encoder_embedding, self.enc_dec_model.enc_dec_model.encoder)
+
+    @property
+    def decoder(self):
+        return DecWEmb(self.enc_dec_model.decoder_embedding, self.enc_dec_model.enc_dec_model.decoder)
+
+    @property
+    def tokens_head(self):
+        return TokensHeadWEmb(self.enc_dec_model.decoder_embedding, self.enc_dec_model.tokens_head)
```

Comment:
Shouldn't these be going into a base class? Also, can you consider changing the suffix `WEmb` Maybe just remove the `W`?